libbeat/cfgfile/reload.go

```diff
@@ -204,21 +212,22 @@ func (rl *Reloader) Run(runnerFactory RunnerFactory) {
 				logp.Err("Error fetching new config files: %v", err)
 			}
 
-			// no file changes
-			if !updated && !overwriteUpdate {
-				overwriteUpdate = false
+			// if there are no changes, skip this reload unless forceReload is set.
+			if !updated && !forceReload {
 				continue
 			}
+			configReloads.Add(1)
```

Comment:
Wondering if the config reload counter should only increase when it was successful? And have an additional error counter for debugging?

Reply:
I'm hesitant, because an error counter here wouldn't be at the right granularity -- we get an error here even if almost everything reloads successfully and there's one invalid file in the search path, and it would be misleading to report that as though the reload didn't happen.

The related monitor vars `libbeat.config.module.{starts, stops, running}` are defined in this file also but they aren't used at all. I was planning to enable them in a followup (it requires changing the `RunnerList` API and modifying other call sites or I'd have done it in this PR), but it might make sense to add `errors` at the same time to track reload failures at  module granularity instead of the whole configuration directory.